src/tplr/neurons.py

```diff
@@ -533,3 +534,24 @@ def pack_binary_tensor(tensor: torch.Tensor, device: DeviceLikeType):
         packed_tensor |= tensor[i::8] << i  # Pack 8 values per byte
 
     return packed_tensor
+
+
+async def periodic_registration_check(
```

Comment:
this is more of a concern for the miner rather than validators. So I wouldn't consider this as a core-common functionality to be honest.

And even for miner, it'd need to be an optional, as it'll take CPU & network time that miners need more for the core process of training.